wpilibj/src/main/java/edu/wpi/first/wpilibj/DriverStation.java

```diff
@@ -1106,12 +1111,35 @@ private void reportJoystickUnpluggedError(String message) {
    */
   private void reportJoystickUnpluggedWarning(String message) {
     double currentTime = Timer.getFPGATimestamp();
-    if (currentTime > m_nextMessageTime) {
+    if (!m_silenceJoystickWarnings && currentTime > m_nextMessageTime ) {
```

Comment:
I'd like to try to do something here to print the warning once and then stop when silenced is true.  Maybe silence is the wrong word here.  Maybe we move the if to be around the line `m_nextMessageTime = currentTime + JOYSTICK_UNPLUGGED_MESSAGE_INTERVAL;`  And then set nextMessageTime to be something that will never be called (maybe -1 or infinity)?  The setter would need to reset this time.

Reply:
I like the idea of sending it only once, making it less of a "silence" but rather a spam suppression. I'll poke around on that.